jsftemplating/src/main/java/com/sun/jsftemplating/util/fileStreamer/ResourceContentSource.java

```diff
@@ -122,11 +122,8 @@ public static String normalize(String origPath) {
 	// Normalize it...
 	if ((path != null) && (path.length() > 0)) {
 	    path = path.replace('\\', '/');
-            // Remove leading '/' chars
-	    while ((path.length() > 0) && (path.charAt(0) == '/')) {
-		path = path.substring(1);
-	    }
```

Comment:
I think it is expected that leading '/' characters are stripped during this method.  You could move the stripping of leading '/''s to the end of this method instead of doing it 1st.  And you could add a leading '/' at the beginning to ensure it always has one... that might work.  I suspect w/ this latest the path `../bad/path` would fail to throw an exception.

Also, I recommend that you check in these test cases (I can submit a PR if you'd like). This way it will ensure these cases always get considered when making changes.

Reply:
I opened a PR to your PR which adds a few tests.